mh/pat.py

```diff
@@ -1804,7 +1829,14 @@ def sendNtcFriendAccept(self, recipient_id, info, message, seq):
         info.unk_long_0x02 = pati.Long(20)
         data += info.pack()
         data += pati.lp2_string(message)
-        self.send_packet(PatID4.NtcFriendAccept, data, seq)
+
+        users = self.session.find_users(recipient_id, "", 1, 1)
+        assert len(users)>0, "User is offline or does not exist."
```

Comment:
This sounds like a bad idea, a user sending a friend request shouldn't be disconnected because the other user is offline or doesn't exist.

Reply:
Ditto to above, but I'm more inclined to think that you cannot send messages/friend requests to offline users at all, and it should instead just not send if the recipient is offline. Otherwise we would need to write some sort of mail queue system into the database.

Reply:
When the user is not online or doesn't exist you can reply with a [AnsAlert](https://github.com/sepalani/MH3SP/blob/dev/mh/pat.py#L145) with the proper message to notify the player of the situation.

Reply:
Done